### DISCUSSION TITLE
Speed issues

### DISCUSSION THREAD

**Discussion by lesquoyb** [gama-platform/gama#313] [General]
**Is your request related to a problem? Please describe.**
I'm creating this issue to track general speed issues in gama. If you have particular examples of gama being slower/faster in certain situation please contribute to this issue as well. So far I have identify three problems that may be great paths to follow to gain in general simulation speed:

1. In some models **without display**, running headless is massively faster than gui, up to two times faster. I haven't identified what exactly is the problem yet as some models have little to no difference between gui and headless.
2. In gui mode, on windows (not tested on other platforms yet), having the computer focused on another window (like web browser) lead to consistent performance drop. From 20 to 50 percent on models I have checked.
3. repeating an experiment improves its performances significantly, I also manage to bring it to 2 times faster on some models after repeating 10 times. Though this one makes sense because the jvm as well as the cpu are going to optimize repeated calls over time, seeing that we can decrease speed by 50% makes me think that there are some gains to grab here as well by caching functions ourselves or something like this.

**Comment by AlexisDrogoul** [comment]
I think this should better belong to a discussion, right ? I'm converting it.
